FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClientFactory.cs

```diff
@@ -30,15 +35,24 @@ public IAzureBlobContainerClient Create(string storeUrl, string containerName)
             // Create client for local emulator.
             if (string.Equals(storeUrl, StorageEmulatorConnectionString, StringComparison.OrdinalIgnoreCase))
             {
-                return new AzureBlobContainerClient(StorageEmulatorConnectionString, containerName, _loggerFactory.CreateLogger<AzureBlobContainerClient>());
+                return CreateUsingConnectionString(StorageEmulatorConnectionString, containerName);
             }
 
             var storageUri = new Uri(storeUrl);
             var containerUrl = new Uri(storageUri, containerName);
 
             return new AzureBlobContainerClient(
                 containerUrl,
+                _credentialProvider,
                 _loggerFactory.CreateLogger<AzureBlobContainerClient>());
         }
+
+        public IAzureBlobContainerClient CreateUsingConnectionString(string connectionString, string containerName)
```

Comment:
is it ok to mark this function private?

Reply:
We'd like to add an interface for connection string to create a client.